test/parallel/test-process-exit-code-validation.js

```diff
@@ -141,3 +138,10 @@ if (process.argv[2] === undefined) {
     process.exit(args[index].code);
   }
 }
+
+const origExitCode = Object.getOwnPropertyDescriptor(process, 'exitCode');
+try {
+  delete process.exitCode;
```

Comment:
I think we should just test that redefining `process.exitCode` with accessors that call out to the original at the end still works as expected. That means users can still mock it if they call out to the original accessors. If they don't, or if they delete it, then it is just not going to get picked up as the actual exit code. And we can just put a note in the doc warning that `delete process.exitCode` is not going to work, like what @isaacs suggested in https://github.com/nodejs/node/pull/49579#issuecomment-1712870183